### PR TITLE
Enable rendering of nested components

### PR DESCRIPTION
**Pull request by jklina**
This alters how slots are rendered on the view story page. The main change has the slots render themselves rather then evaluating the contents of the slots' block outside of the slots' context. This should allow stories to render nested slots which I don't believe is currently possible.

### CODE REVIEW

File: spec/view_component/storybook/stories_controller_spec.rb [hunk 1]
```diff
@@ -151,9 +151,9 @@
     expect(response.body).to have_selector(".tab", text: "Tab A")
     expect(response.body).to have_selector(".tab", text: "Tab B")
 
-    expect(response.body).to have_selector(".item", count: 3)
-    expect(response.body).to have_selector(".item.highlighted", count: 1)
-    expect(response.body).to have_selector(".item.normal", count: 2)
+    expect(response.body).to have_selector(".item", count: 4)
+    expect(response.body).to have_selector(".item", text: "Subslot", count: 1)
+    expect(response.body).to have_selector(".item.normal", count: 3)
```
Comment (jonspalmer) [inline spec/view_component/storybook/stories_controller_spec.rb:156]: Why does this change end ip with more items?

File: spec/view_component/storybook/stories_controller_spec.rb [hunk 2]
```diff
@@ -151,9 +151,9 @@
     expect(response.body).to have_selector(".tab", text: "Tab A")
     expect(response.body).to have_selector(".tab", text: "Tab B")
 
-    expect(response.body).to have_selector(".item", count: 3)
-    expect(response.body).to have_selector(".item.highlighted", count: 1)
-    expect(response.body).to have_selector(".item.normal", count: 2)
+    expect(response.body).to have_selector(".item", count: 4)
+    expect(response.body).to have_selector(".item", text: "Subslot", count: 1)
```
Comment (jonspalmer) [inline spec/view_component/storybook/stories_controller_spec.rb:155]: We should keep the test of `highlighted`

**Comment by jonspalmer** [comment]
I'd like to understand a slightly larger use case than is shared in the tests. I have concern that this approach isn't going to work in some cases. The reason we do `instance_eval` when rendering content_block (for the component or the slots) is that it allows the correct rendering of slots that use various view helpers etc. I think we'd want that to be true for nested components too?

**Comment by dark-panda** [comment]
@jklina @jonspalmer I believe you'd be able to combine both the use of helpers with nested components using the following addition to this PR:

https://github.com/dark-panda/view_component_storybook/commit/ec07f4261d16e6608d3562f7808819056358e8b7

**Comment by jonspalmer** [comment]
@jklina @dark-panda I've added a slightly deeper example to the specs and included the suggested patch. I think we now get nested slots and helpers. Please take a look and see if these supports the use cases you'd expect. Happy to merge and release a new version if it does.

**Comment by jklina** [comment]
This looks great! Thank you and @dark-panda for taking this over the finish line! 👊

**Comment by dark-panda** [comment]
Cheers, I'll give this a shot soon to see how it looks on my end.

**Pull request closed without merging**